content/react.md

```diff
@@ -0,0 +1,271 @@
+
+(Note that "container components" are analogous to the "smart components" and "presentational components" to the "reusable components" in the pattern we document in the [UI/UX article](http://guide.meteor.com/ui-ux.html#components), if you'd like to read more about how this philosophy marries with Meteor).
+
+For example, in the Todos example app, we have a `ListPage` component, which renders the metadata about a list alongside the todos that are within it. In order to do so, it needs to [subscribe](data-loading.html#subscriptions) to the `todos.inList` publication, check that subscription's readiness, then fetch the list of todos from the `Todos` collection.
+
+It also needs to be responsive to reactive changes in the state of those actions (for instance if a todo changes due to the action of another user). All this data loading complexity is a typical use-case for a container-presentational component split, and the `createContainer()` function makes it simple to do this.
+
+We simply define the `ListPage` component to be a presentational component that expects it's data to be passed in as a property:
+
+```jsx
+import React from 'react';
+
+export default class ListPage extends React.Component {
+  ...
+}
+
+ListPage.propTypes = {
+  list: React.PropTypes.object,
+  todos: React.PropTypes.array,
+  loading: React.PropTypes.bool,
+  listExists: React.PropTypes.bool,
+};
+```
+
+Then we create a `ListContainer` component which wraps it and provides a data source:
+
+```jsx
+import { Meteor } from 'meteor/meteor';
+import { Lists } from '../../api/lists/lists.js';
+import createContainer from 'meteor/react-meteor-data';
+import ListPage from '../pages/ListPage.jsx';
+
+export default createContainer(({ params: { id } }) => {
+  const todosHandle = Meteor.subscribe('todos.inList', id);
+  const loading = !todosHandle.ready();
+  const list = Lists.findOne(id);
+  const listExists = !loading && !!list;
+  return {
+    loading,
+    list,
+    listExists,
+    todos: listExists ? list.todos().fetch() : [],
+  };
+}, ListPage);
+```
+
+Note that the `MeteorDataContainer` container created by `createContainer()` will be fully reactive to any changes to [reactive data sources](https://atmospherejs.com/meteor/tracker) call from inside the function provided to it.
+
+<h3 id="preventing-rerenders">Preventing re-renders</h3>
+
+Sometimes the Tracker system can lead to unnecessary re-computations, which when combined with `react-meteor-data` leads to unnecessary re-renders. Although React is in general quite efficient in such cases, if you need to control re-rendering, the above pattern allows you to easily use React's [`shouldComponentUpdate`](https://facebook.github.io/react/docs/component-specs.html#updating-shouldcomponentupdate) on the presentational component wrapped and avoid re-renders.
+
+<h2 id="routing">Routing</h2>
```

Comment:
Worth linking to the routing guide up front to say "here are some principles about routing"